charts/steward/templates/clusterrole-run-controller.yaml

```diff
@@ -0,0 +1,37 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: steward-run-controller
+  labels:
+    {{- include "steward.labels" . | nindent 4 }}
+rules:
+- apiGroups: ["rbac.authorization.k8s.io"]
+  resources: ["rolebindings"]
+  verbs: ["get","list","create"]
+- apiGroups: ["rbac.authorization.k8s.io"]
+  resources: ["clusterroles"]
+  verbs: ["bind","get"]
+  resourceNames: ["steward-run"]
+- apiGroups: ["networking.k8s.io"]
+  resources: ["networkpolicies"]
+  verbs: ["create"]
+- apiGroups: [""]
+  resources: ["serviceaccounts"]
+  verbs: ["get","list","create","update"]
+- apiGroups: ["steward.sap.com"]
+  resources: ["pipelineruns","pipelineruns/status"]
+  verbs: ["get","list","patch","update","watch"]
+- apiGroups: ["tekton.dev"]
+  resources: ["taskruns"]
+  verbs: ["create","delete","get","list","patch","update","watch"]
+- apiGroups: [""]
+  resources: ["namespaces","secrets","resourcequotas","limitranges"]
+  verbs: ["create","delete","get","list","patch","update","watch"]
+## may be restricted to steward-system namespace???
```

Comment:
👍 did not check if more is required, but sounds better. In general we should restrict the permissions to only the required namespaces/resources.

Reply:
> In general we should restrict the permissions to only the required namespaces/resources.

Works only if statically known. Dynamically generated namespaces cannot be covered, as K8s RBAC doesn't support more flexible rules (most likely due to security concerns).